catalog/async/catalog-async-data/src/main/java/org/codice/ddf/catalog/async/data/impl/LazyProcessResourceImpl.java

```diff
@@ -111,6 +114,21 @@ public String getName() {
     return name;
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * <p>This will load the resource.
+   */
+  @Override
+  public byte[] getByteArray() throws IOException {
```

Comment:
❗️ Do we really want to return a reference to the internal byte array? Changes to that will impact the input stream and this resource implementation. How about a ByteBuffer in read-only mode instead?